### ISSUE TITLE
cannot upload files

### ISSUE THREAD

**Issue by simllll** [cloudconvert/cloudconvert-node#53]
I do following things:
create a new job
```ts
	const job = await cloudConvert.jobs.create({
				tasks: {
					'upload-my-file': {
						operation: 'import/upload'
					},
					'convert-my-file': {
						operation: 'convert',
						input: ['upload-my-file'],
						input_format: inputformat,
						output_format: outputformat
						// timeout: 60
					},
					'export-my-file': {
						operation: 'export/url',
						input: ['convert-my-file']
					}
				}
			});
```

upload file
```ts
			const uploadTask = job.tasks.filter(task => task.name === 'upload-my-file')[0];
			const readable = streamifier.createReadStream(fileContent);

			await cloudConvert.tasks.upload(uploadTask, readable);

```

This is sent to https://storage.de.cloud.ovh.net/v1/AUTH_b2cffe8f45324c2bba39e8db1aedb58f/cloudconvert-files/823bfe12-0dc3-4356-991b-b95070cd7101/ but the API always returns a redirect like this:

      path: '/v2/upload/redirect/823bfe12-0dc3-4356-991b-b95070cd7101?status=400&message=no%20files%20to%20process',

it says "no files to process", but not quite sure why. There is a file, the readbable is also perfectly fine:

```
readable MultiStream {
  _readableState: ReadableState {
    objectMode: false,
    highWaterMark: 16384,
    buffer: BufferList { head: null, tail: null, length: 0 },
    length: 0,
    pipes: null,
    pipesCount: 0,
    flowing: null,
    ended: false,
    endEmitted: false,
    reading: false,
    sync: true,
    needReadable: false,
    emittedReadable: false,
    readableListening: false,
    resumeScheduled: false,
    emitClose: true,
    autoDestroy: false,
    destroyed: false,
    defaultEncoding: 'utf8',
    awaitDrainWriters: null,
    multiAwaitDrain: false,
    readingMore: false,
    decoder: null,
    encoding: null,
    [Symbol(kPaused)]: null
  },
  readable: true,
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  _object: <Buffer 50 4b 03 04 14 00 08 08 08 00 8a 5b a2 50 00 00 00 00 00 00 00 00 00 00 00 00 12 00 00 00 77 6f 72 64 2f 6e 75 6d 62 65 72 69 6e 67 2e 78 6d 6c a5 93 ... 190106 more bytes>,
  [Symbol(kCapture)]: false
}
```

**Comment by simllll** [comment]
I've updated to latest version which inlcudes TS definitions, now I'm unsure why upload needs a Task and not a JobTask as input? Is this a mistake?

**Comment by josiasmontag** [comment]
Does the same happen if you are using a file stream?
```js
const readable = fs.createReadStream('./file.pdf');
```

**Comment by simllll** [comment]
It does, but I don't get why.
I tested the upload form with a local endpoint, the file is transmitted correctly in both ways. Is there any kind of check on cloudconvert side that prevents this way of uploading a file? I tried setting an explicit filename, but didn't change anything.

**Comment by simllll** [comment]
okay super strange, I'm working in a monorepo and it seems there was another version of the lib installed. I removed the newer dependecy and now it started working. At least for the upload part, but now the final document is just two empty white pages?

**Comment by simllll** [comment]
It stopped working again, I didn't change any single line of code, just ensured that the latest version of cloudconvert lib is installed. Downgraded to 2.0, but doesn't solve this issue either. It must be some kind of dependency. I have form-data v3 installed, looking it it now if this is related.

**Comment by simllll** [comment]
I don't get it, did something change on the serverside in the meantime? 
I had a running version, but it stopped working without any code change.
* Tested cloudconvert 2.0.0, 2.1.2
* ensured that form-data is set to version 2.5

I always get  a redirect to /v2/upload/redirect/f0ad7272-a9a4-4336-acd5-5c82d90e9247?status=400&message=no%20files%20to%20process

Any hints what else I could try?

**Comment by simllll** [comment]
Alright, I finally found a solution.
I have to provide a filename as 3rd argument to formData.append

`formData.append("file", stream, filename);`

this also solves my issue that I have to convert the buffer to a stream first, I can now directly pass the buffer to the formdata. It works with a stream but now too.

**Comment by KnorpelSenf** [comment]
> I've updated to latest version which inlcudes TS definitions, now I'm unsure why upload needs a Task and not a JobTask as input? Is this a mistake?

You might be right with this. This should be a union type, both should be permitted.

**Comment by KnorpelSenf** [comment]
@simllll I submitted a PR for the typing problem with job tasks

**Comment by josiasmontag** [comment]
Should be fixed by version 2.1.3
